src/django/api/management/commands/batch_process.py

```diff
@@ -0,0 +1,104 @@
+                            geocode_facility_list_item,
+                            match_facility_list_item)
+
+ACTIONS = {
+    ProcessingAction.PARSE: parse_facility_list_item,
+    ProcessingAction.GEOCODE: geocode_facility_list_item,
+    ProcessingAction.MATCH: match_facility_list_item,
+}
+
+
+class Command(BaseCommand):
+    help = 'Run an action on all items in a facility list. If ' \
+           'AWS_BATCH_JOB_ARRAY_INDEX environment variable is set, will ' \
+           'process those items whose row_index matches it. Otherwise, will ' \
+           'process all items for the given facility list.'
+
+    def add_arguments(self, parser):
+        # Create a group of arguments explicitly labeled as required,
+        # because by default named arguments are considered optional.
+        group = parser.add_argument_group('required arguments')
+        group.add_argument('-a', '--action',
+                           required=True,
+                           help='The processing action to perform. '
+                                'One of "parse", "geocode", "match"')
+        group.add_argument('-l', '--list-id',
+                           required=True,
+                           help='The id of the facility list to process.')
+
+    def handle(self, *args, **options):
+        action = options['action']
+        list_id = options['list_id']
+
+        try:
+            if action not in ACTIONS.keys():
+                raise ValueError('Invalid action "{}". Must be one of '
+                                 '"parse", "geocode", "match".'.format(action))
+
+            process = ACTIONS[action]
+            facility_list = FacilityList.objects.get(pk=list_id)
+
+            row_index = os.environ.get('AWS_BATCH_JOB_ARRAY_INDEX')
+            if row_index:
+                items = FacilityListItem.objects.filter(
+                    facility_list=facility_list,
+                    row_index=row_index)
+            else:
+                items = FacilityListItem.objects.filter(
+                    facility_list=facility_list)
+
+            result = {
+                'success': 0,
+                'failure': 0,
+            }
+
+            # Process all items, save affected items, facilities, matches,
+            # and tally successes and failures
+            for item in items:
+                try:
+                    if action == ProcessingAction.MATCH:
+                        facility, match = process(item)
+                        with transaction.atomic():
+                            item.save()
+                            facility.save()
+                            match.save()
+                    else:
+                        process(item)
+                        item.save()
+
+                    if item.status == FacilityListItem.ERROR:
+                        result['failure'] += 1
+                    else:
+                        result['success'] += 1
+                except DatabaseError:
```

Comment:
I'm catching a `DatabaseError` here because that's what `transaction.atomic()` will throw. `process` has error handlers inside it, and will catch any local errors.